### PR TITLE
Updates for Label Only membership inference attack

### PR DESCRIPTION
**Pull request by beat-buesser**
Signed-off-by: Beat Buesser <beat.buesser@ie.ibm.com>

# Description

This pull request provides updates for the Label Only Boundary Distance membership inference attack.

## Type of change

Please check all relevant options.

- [x] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: art/attacks/inference/membership_inference/label_only_boundary_distance.py [hunk 1]
```diff
@@ -70,6 +71,18 @@ def infer(self, x: np.ndarray, y: Optional[np.ndarray] = None, **kwargs) -> np.n
         """
         from art.attacks.evasion.hop_skip_jump import HopSkipJump
 
+        if y is None:
```
Comment (abigailgold) [inline art/attacks/inference/membership_inference/label_only_boundary_distance.py:74]: Why not make the parameter non-optional?

File: art/attacks/inference/membership_inference/label_only_boundary_distance.py [hunk 2]
```diff
@@ -70,6 +71,18 @@ def infer(self, x: np.ndarray, y: Optional[np.ndarray] = None, **kwargs) -> np.n
...
+            raise ValueError("Argument `y` is None, but this attack requires true labels `y` to be provided.")
+
+        if self.distance_threshold_tau is None:
+            raise ValueError(
+                "No value for distance threshold `distance_threshold_tau` provided. Please set"
+                "`distance_threshold_tau` or run method `calibrate_distance_threshold` on known training and test"
+                "dataset."
+            )
+
+        y = check_and_transform_label_format(y, self.estimator.nb_classes)
```
Comment (abigailgold) [inline art/attacks/inference/membership_inference/label_only_boundary_distance.py:84]: Call to check_and_transform_label_format should also be added to calibrate_distance_threshold method (both on y_train and y_test)

Comment (beat-buesser) [reply art/attacks/inference/membership_inference/label_only_boundary_distance.py:98]: I agree and have updated the code.

File: art/attacks/inference/membership_inference/label_only_boundary_distance.py (hunk 1 above)
Comment (beat-buesser) [reply art/attacks/inference/membership_inference/label_only_boundary_distance.py:82]: This to follow the overall pattern in ART. For membership inference attacks we could discuss to remove the Optional in general as it is not supported by any of the current membership inference attacks.

File: art/attacks/inference/membership_inference/label_only_boundary_distance.py (hunk 2 above)
Comment (abigailgold) [reply art/attacks/inference/membership_inference/label_only_boundary_distance.py:98]: Ok. Looks good.

**Pull request merged**